### DISCUSSION TITLE
(PWA) Must have support for Web Applications. Very important! The Non-Negotiable Game-Changer for Modern Browsers

### DISCUSSION THREAD

**Discussion by Ledenav** [zen-browser/desktop#1415] [Ideas]
# PWA Support: The Future of Web Browsing

![PWA Illustration](https://i.pcmag.com/imagery/articles/05ULMhCB6uHHlEQdWyy5V2Z-1..v1629405588.jpg)

## Key Benefits

| For Users | For Developers | For Browsers |
|-----------|----------------|--------------|
| Native app-like experience | Platform-independent development | Increased competitiveness |
| Fast loading times | Offline functionality | Future-proof technology |
| Enhanced security | Efficient resource use | Broader web ecosystem |

> PWAs bridge the gap between websites and native apps, offering a richer, more accessible internet experience.

**Embracing PWA support isn't just an option—it's a necessity for modern browsers.**

**Comment by Muko-Tabi** [comment]
Could try this [extension](https://addons.mozilla.org/en-US/firefox/addon/pwas-for-firefox) for now, but it doesn't work in Flatpak for [some](https://github.com/filips123/PWAsForFirefox/issues/76#issuecomment-962628218) [reason](https://github.com/flatpak/xdg-desktop-portal/pull/705)

**Reply by cyrneko** [reply]
> Could try this [extension](https://addons.mozilla.org/en-US/firefox/addon/pwas-for-firefox) for now, but it doesn't work in Flatpak for [some](https://github.com/filips123/PWAsForFirefox/issues/76#issuecomment-962628218) [reason](https://github.com/flatpak/xdg-desktop-portal/pull/705)

TL;DR: There is no way to access the Native Messaging API over Portals (APIs primarily used by Flatpaks to access things outside of the sandbox) and as such, nothing that relies on it will work out-of-the-box.

There is some proposals to implement such a portal [here](https://github.com/flatpak/xdg-desktop-portal/issues/655#issue-1050200719), but the proposal is not finished (?) and no implementations exist.

Although, the NativeMessaging API is not necessarily required to implement PWAs, as seen by Chromium-Based browsers doing it under Flatpak and Snap respectively.

**Comment by Ledenav** [comment]
We must bring this issue to the agenda!
